Shared/Tests.Shared/TestObjects.cs

```diff
@@ -44,6 +44,9 @@ public class AllTypesObject : RealmObject
 
         public DateTimeOffset DateTimeOffsetProperty { get; set; }
 
+        [Required]
+        public string RequiredStringProperty { get; set; }
```

Comment:
Just to double-check we have a type that is actually passed to a live realm so that the C++ schema code can exercise and validate it. The `ObjectSchemaTests` fixture doesn't actually touch the C++ schema logic.